designer/server/src/test/scala/pl/touk/nussknacker/ui/process/deployment/DeploymentServiceSpec.scala

```diff
@@ -78,14 +79,18 @@ class DeploymentServiceSpec
 
   private val processingTypeDataProvider: ProcessingTypeDataProvider[DeploymentManager, Nothing] =
     new ProcessingTypeDataProvider[DeploymentManager, Nothing] {
-      override def forType(processingType: ProcessingType)(implicit user: LoggedUser): Option[DeploymentManager] =
-        all.get(processingType)
 
-      override def all(implicit user: LoggedUser): Map[ProcessingType, DeploymentManager] = Map(
-        TestProcessingTypes.Streaming -> deploymentManager
-      )
+      override val state: ProcessingTypeDataState[DeploymentManager, Nothing] =
+        new ProcessingTypeDataState[DeploymentManager, Nothing] {
```

Comment:
ah, the cursed var ;) TBH we could use e.g. `Supplier[DeploymentManager]` instead of using var. To loosen this coupling. But it's up to you. In the end, it's not such a big deal.

Reply:
I'll leave it as it is. It would be complicated with this `Supplier`. We'd have to change provided type to `DeploymentService`